mode-apis-base/src/entityModels.ts

```diff
@@ -343,6 +343,10 @@ export interface EntityAttributeParam<ValueType extends EntityAttributeValueType
     readonly min?: number | undefined;
     readonly max?: number | undefined;
     readonly regex?: string | undefined;
+
+    // Whether or not the entity can be sorted by this attribute. This is OPTIONAL because existing entity classes' attribute schema
+    // might not have this settings. The default will be FALSE unless this attribute is specifically set to true.
+    readonly sortable?: boolean | undefined;
```

Comment:
@bananaumai and @jamesbalcombe83, is this still required for the attribute to be sortable? Or can the back end sort any attribute whether or not the attribute schema has this flag set to true?

The reason I asked is because there is a scenario which the table show multiple entities of different class. Both classes has the same attribute name but one of the class has this attribute's `sortable` set to true but the other one doesn't. So what would happen if we try to sort by this attribute? For example

## Table of robots
```
Name.                Class                   Serial Number
Robot 1                A                        827438
Robot 2                A                        544343
Robot 3                B                        342345
```

The `Serial Numer` attribute for class A has `sortable` set to true but not for class B.

Reply:
I haven't made use of `sortable` in the backend, anything can be sorted from the db perspective. My assumption was the sortable was there for the frontend to allow/disallow selection for sorting.

If a query included entities that didn't have the field being sorted on, those entities are always included **after** the entities which include the field, whether sorting by ascending or descending order.

Reply:
We discussed this in a meeting and I heard we needed this because it was inefficient for the back end to have all attributes sortable therefore we needed this attribute so that the back end can optimize.

But if all attributes can be sorted then we don't need this in the attribute schema. This can just be EntityUIComponent's customization. Ok, I will remove this.

Reply:
This may be something I am unaware of @bananaumai will be able to confirm.

Reply:
We can define `array` and `object` type entity attribute fields.

So, it's difficult/meaningless for the backend to sort items with those fields.

Implementation-wise, I think if the types are primitive types such as string, number, and boolean, we can assume they are sortable.

@bill-mode So, we don't need the `sortable` field because we can infer from the attribute type if we can use an attribute field for sorting items. 

@jamesbalcombe83 If possible, the backend should verify the attribute field to be specified.

Reply:
Ok. I will have the front end NOT allow attributes of type `Array` and `Object` to be sortable.